File: Bookstore.API/Models/Order.cs
using System;
using System.Collections.Generic;
using Bookstore.API.Data;

namespace Bookstore.API.Models
{
    public class Order
    {
        public int Id { get; set; }
        public ICollection<OrderBook> Books { get; set; }
        public string Status { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Street { get; set; }
        public string HomeNumber { get; set; }
        public string PostNumber { get; set; }
        public string City { get; set; }

        public string PaymentMethod { get; set; }
        public DateTime? ShippingDate { get; set; }
        public DateTime OrderDate { get; set; }
        public float TotalPrice { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }

    }
}

File: Bookstore-SPA/src/app/_models/order.ts
import { OrderBook } from './order-book';

export class Order {
    id: number;
    books: OrderBook[];
    status: string;
    paymentMethod: string;
    shippingDate?: Date;
    orderDate: Date;
    totalPrice: number;
    street: string;
    homeNumber: string;
    postNumber: string;
    city: string;
}
